### PR TITLE
API to add ProductDetail and ProductImage

### PR DESCRIPTION
**Pull request by leminhhuan72**
Description:
 - API to add a ProductDetail to an existed Product
 - API to add a ProductImage to an existed Product

### CODE REVIEW

File: src/main/java/com/huan/ecommerce/service/impl/ProductDetailService.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        Long productId = productDetailDTO.getProductId();
+        if(!productRepository.existsById(productId.intValue())) {
+            throw new EntityNotFoundException("Product does not exists. Product ID: " + productId);
+        }
+        if (productDetailRepository.existsProductDetailByProductId(productId)) {
+               throw new EntityExistsException("Product detail with the given product_id already exists. Product ID: " + productId);
+        }
+        Product product = productRepository.getReferenceById(productDetailDTO.getProductId().intValue());
+        ProductDetail productDetail = EntityDTOMapper.mapProductDetailDTOToProductDetail(productDetailDTO);
+        productDetail.setProduct(product);
```
Comment (gdragon0598) [inline src/main/java/com/huan/ecommerce/service/impl/ProductDetailService.java:37]: how about store list images of product?

File: src/main/java/com/huan/ecommerce/service/impl/ProductDetailService.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    private ProductDetailRepository productDetailRepository;
+    /**
+     *
+     * @param productDetailDTO
+     * @return
+     */
+    @Override
+    public ProductDetailDTO saveProduct(ProductDetailDTO productDetailDTO) {
+        Long productId = productDetailDTO.getProductId();
+        if(!productRepository.existsById(productId.intValue())) {
```
Comment (gdragon0598) [inline src/main/java/com/huan/ecommerce/service/impl/ProductDetailService.java:29]: id will auto generate... so restrict user passed id in request body and don't need check.. improve performance

File: src/main/java/com/huan/ecommerce/controller/ProductDetailController.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+@RestController
+@RequestMapping("/product-detail")
+public class ProductDetailController {
+    @Autowired
+    private IProductDetailService productDetailService;
+
+    @PostMapping
+    public String addProductDetail(@RequestBody @Valid ProductDetailDTO productDetailDTO) {
+        productDetailService.saveProduct(productDetailDTO);
+        return "Product detail added successfully";
```
Comment (gdragon0598) [inline src/main/java/com/huan/ecommerce/controller/ProductDetailController.java:18]: `Return: ResponseEntity<ProductionDetailDto>`

File: src/main/java/com/huan/ecommerce/controller/ProductDetailController.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    @PutMapping
+    public String updateProductDetail(@RequestBody @Valid ProductDetailDTO productDetailDTO) {
+        productDetailService.updateProduct(productDetailDTO);
+        return "Product detail updated successfully";
```
Comment (gdragon0598) [inline src/main/java/com/huan/ecommerce/controller/ProductDetailController.java:24]: `Return: ResponseEntity<ProductionDetailDto>`

File: src/main/java/com/huan/ecommerce/service/impl/ProductDetailService.java (hunk 2 above)
Comment (leminhhuan72) [reply src/main/java/com/huan/ecommerce/service/impl/ProductDetailService.java:29]: product id is the id of the product not the product detail. Here, the idea is to let user add Product and Product Detail separately

Comment (gdragon0598) [reply src/main/java/com/huan/ecommerce/service/impl/ProductDetailService.java:29]: no in admin dashboard only 1 action for product.... 1 action for store product, product detail , also list images

Comment (gdragon0598) [reply src/main/java/com/huan/ecommerce/service/impl/ProductDetailService.java:29]: and remember that if occur any error please rollback transaction

**Comment by leminhhuan72** [comment]
Mr Tri reviewed.

**Pull request merged**